test/iidm/ThreeWindingsTransformerTest.cpp

```diff
@@ -329,6 +403,52 @@ BOOST_AUTO_TEST_CASE(constructor) {
     BOOST_TEST(stdcxx::areSame(cTerminal3, cLeg3.getTerminal()));
     BOOST_TEST(stdcxx::areSame(network, leg3.getNetwork()));
     BOOST_TEST(stdcxx::areSame(network, cLeg3.getNetwork()));
+
+    const auto& legs = transformer.getLegs();
+    BOOST_CHECK_EQUAL(3, boost::size(legs));
+    int index = 0;
+    for (const auto& leg : legs) {
+        switch (index) {
+            case 0:
+                BOOST_CHECK(stdcxx::areSame(leg1, leg));
+                break;
+
+            case 1:
+                BOOST_CHECK(stdcxx::areSame(leg2, leg));
+                break;
+
+            case 2:
+                BOOST_CHECK(stdcxx::areSame(leg3, leg));
+                break;
+
+            default:
+                break;
+        }
+        index++;
+    }
+
+    const auto& cLegs = cTransformer.getLegs();
+    BOOST_CHECK_EQUAL(3, boost::size(cLegs));
+    index = 0;
+    for (const auto& leg : cLegs) {
+        switch (index) {
+            case 0:
+                BOOST_CHECK(stdcxx::areSame(leg1, leg));
+                break;
+
+            case 1:
+                BOOST_CHECK(stdcxx::areSame(leg2, leg));
+                break;
+
+            case 2:
+                BOOST_CHECK(stdcxx::areSame(leg3, leg));
+                break;
+
+            default:
+                break;
+        }
+        index++;
+    }
```

Comment:
In java, we have something to compare to iterables. Could you check if boost can also do it (BOOST_CHECK_EQUAL_COLLECTIONS) ?

Reply:
`BOOST_CHECK_EQUAL_COLLECTION` does not take any predicate, and we have to provide one to compare legs.
The best way I found to do that is 
```
    std::vector<std::reference_wrapper<ThreeWindingsTransformer::Leg>> ref = {leg1, leg2, leg3};
    BOOST_CHECK(std::equal(legs.begin(), legs.end(), ref.begin(), [](const ThreeWindingsTransformer::Leg& l1, const ThreeWindingsTransformer::Leg& l2) {
        return stdcxx::areSame(l1, l2);
    }));
    ```

Reply:
Ok fixed by adding an utility to wrap STL : 
```
    POWSYBL_ASSERT_COLLECTIONS_EQUAL(expected, transformer.getLegs(), legComparator);
```